### PR TITLE
IC-1853: Add probation practitioner referral details page

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
## What does this pull request do?

Adds a referral details page for probation practitioners (reusing the existing page that we have for service providers), and removes the UI for assigning a referral (for PPs only).

## What is the intent behind these changes?

To allow PPs to view the details of a referral they've submitted.

### CODE REVIEW

File: server/routes/shared/showReferralPresenter.test.ts [hunk 1]
```diff
@@ -68,7 +92,15 @@ describe(ShowReferralPresenter, () => {
...
     it('returns a summary list of probation practitioner details', () => {
       const sentReferral = sentReferralFactory.build(referralParams)
-      const presenter = new ShowReferralPresenter(sentReferral, intervention, deliusUser, null, null)
+      const presenter = new ShowReferralPresenter(
+        sentReferral,
+        intervention,
+        deliusUser,
+        null,
+        null,
+        'service-provider',
```
Comment (tomsmyers) [inline server/routes/shared/showReferralPresenter.test.ts:101]: nit: this is the wrong prefix for this test

File: server/routes/shared/showReferralPresenter.test.ts [hunk 2]
```diff
@@ -37,7 +37,15 @@ describe(ShowReferralPresenter, () => {
   describe('assignmentFormAction', () => {
     it('returns the relative URL for the check assignment page', () => {
       const referral = sentReferralFactory.build(referralParams)
-      const presenter = new ShowReferralPresenter(referral, intervention, deliusUser, null, null)
+      const presenter = new ShowReferralPresenter(
```
Comment (tomsmyers) [inline server/routes/shared/showReferralPresenter.test.ts:40]: would be good to validate that the url changes as the presenter url prefix parameter changes

File: server/routes/shared/referralOverviewPagePresenter.ts [hunk 3]
```diff
@@ -25,4 +25,6 @@ export default class ReferralOverviewPagePresenter {
       },
     ],
   }
+
+  readonly dashboardURL = `/${this.subNavUrlPrefix}/referrals/dashboard`
```
Comment (tomsmyers) [inline server/routes/shared/referralOverviewPagePresenter.ts:29]: would be good to test somewhere that these urls get constructed properly for both sp and pp versions of the page

**Review by tomsmyers — APPROVED** [review]
nice work - thank you!!

File: server/routes/shared/referralOverviewPagePresenter.ts (hunk 3 above)
Comment (lawrence-forooghian) [reply server/routes/shared/referralOverviewPagePresenter.ts:29]: 👍 done

**Pull request merged**